### PR TITLE
[DHCPv6 relay monitor] Add an offset bound check during DHCPv6 header inspection to prevent segmentation fault

### PR DESCRIPTION
**Pull request by shlomibitton**
#### Why I did it
Fixed issue https://github.com/Azure/sonic-buildimage/issues/10326
Adding an offset bound check during DHCPv6 header inspection to prevent segmentation fault issue.

#### How I did it
Each time offset is incremented to fetch a different field in the DHCPv6 header, check if the buffer contains this offset and only then try to fetch it.
If the offset is out of the DHCP header bounds, print a message to syslog and return.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [x] 202106
- [x] 202111

### CODE REVIEW

**Comment by kellyyeh** [comment]
Hi @shlomibitton, did you mean to create the PR for the public Azure sonic-buildimage repo?

**Comment by shlomibitton** [comment]
> Hi @shlomibitton, did you mean to create the PR for the public Azure sonic-buildimage repo?

Hi @kellyyeh, this PR is internal actually and pending review from Nvidia team.
But feel free to review as well :)

File: src/dhcpmon/src/dhcp_device.c [hunk 1]
```diff
@@ -342,19 +342,44 @@ static void read_callback(int fd, short event, void *arg)
...
                 offset += DHCPv6_OPTION_LENGTH + DHCPv6_OPTION_LEN_LENGTH;
+                if (buffer_sz < UDPv6_START_OFFSET + offset)
+                {
+                    syslog(LOG_WARNING, "read_callback(%s): read length (%ld) is too small to capture DHCP options",
+                           context->intf, buffer_sz);
+                    return;
+                }
             }
             handle_dhcpv6_option(context, dhcp_header[offset], dir);
+            return;
```
Comment (yxieca) [inline src/dhcpmon/src/dhcp_device.c:382]: Why is this return needed?

Comment (shlomibitton) [reply src/dhcpmon/src/dhcp_device.c]: It is not, it is a mistake I removed it.

**Review by liorghub — APPROVED** [review]
LGTM

**Pull request closed without merging**